src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -2625,7 +2648,6 @@ mongoc_collection_delete_one (mongoc_collection_t *collection,
                               bson_error_t *error)
 {
    mongoc_delete_one_opts_t delete_one_opts;
-   bson_t limit = BSON_INITIALIZER;
```

Comment:
This was one of the points I brought up in [CDRIVER-4306](https://jira.mongodb.org/browse/CDRIVER-4306), which seemed reasonable to do in this PR.